memo_core/src/epoch.rs

```diff
@@ -1324,6 +1324,11 @@ impl<'a> Cursor<'a> {
         }
     }
 
+    pub fn base_path(&self) -> Result<Option<PathBuf>, Error> {
```

Comment:
Note how, instead of adding a `base_path` field to each `CursorEntry`, we're exposing a method on the cursor. This will make sure users of this API only pay the cost of `base_path` if they need to.